neqo-crypto/src/agent.rs

```diff
@@ -147,22 +150,24 @@ pub struct SecretAgentInfo {
 
 impl SecretAgentInfo {
     fn new(fd: *mut ssl::PRFileDesc) -> Res<Self> {
-        let mut info: ssl::SSLChannelInfo = unsafe { mem::uninitialized() };
+        let mut info: MaybeUninit<ssl::SSLChannelInfo> = MaybeUninit::uninit();
         secstatus_to_res(unsafe {
             ssl::SSL_GetChannelInfo(
                 fd,
-                &mut info,
+                info.as_mut_ptr(),
                 c_uint::try_from(mem::size_of::<ssl::SSLChannelInfo>())?,
             )
         })?;
-        Ok(Self {
-            version: info.protocolVersion as Version,
-            cipher: info.cipherSuite as Cipher,
-            group: Group::try_from(info.keaGroup)?,
-            resumed: info.resumed != 0,
-            early_data: info.earlyDataAccepted != 0,
-            alpn: get_alpn(fd, false)?,
-            signature_scheme: SignatureScheme::try_from(info.signatureScheme)?,
+        Ok(unsafe {
+            Self {
+                version: info.assume_init().protocolVersion as Version,
```

Comment:
It should only be `assumed_init` once.

Reply:
So, I call it once and assign the result to that T and then use that in the resulting Self{}? I don't see that noted anywhere in the docs for `assume_init` which is the only reason I am asking!

Thanks!

Reply:
I see it now. It's moving operation on self.